subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -173,27 +191,43 @@ private String getFilename(Path dir) {
 
         @Override
         public FileVisitResult visitFile(Path file, @Nullable BasicFileAttributes attrs) {
-            String name = stringInterner.intern(file.getFileName().toString());
+            String name = intern(file.getFileName().toString());
             if (isAllowed(file, name, false, attrs, builder.getRelativePath())) {
                 if (attrs == null) {
                     throw new FileSnapshottingException(String.format("Cannot read file '%s': not authorized.", file));
                 }
-                if (attrs.isSymbolicLink()) {
-                    // when FileVisitOption.FOLLOW_LINKS, we only get here when link couldn't be followed
-                    throw new FileSnapshottingException(String.format("Could not list contents of '%s'. Couldn't follow symbolic link.", file));
-                }
-                addFileSnapshot(file, name, attrs);
+                FileSystemLocationSnapshot snapshot = getSnapshot(file, attrs, name);
+                builder.visit(snapshot);
             }
             return FileVisitResult.CONTINUE;
         }
 
+        private FileSystemLocationSnapshot getSnapshot(Path file, BasicFileAttributes attrs, String name) {
```

Comment:
This shouldn't be called `get*()`. It used to be called `snapshot()` which I think is a good name. `createSnapshot()` is also fine if you want to separate the verb and use `snapshot` as a noun.

Reply:
`get` is overloaded and I would omit it in most cases, I agree.
Renamed to `snapshot`